From: ddugger@redhat.com <ddugger@redhat.com>
Date: Wed, 11 Mar 2009 15:44:26 -0600
Subject: [xen] x86: VPID: free resources
Message-id: 20090311214426.GW10195@sobek.n0ano.com
O-Subject: [PATCH 2/2] VPID: free resources
Bugzilla: 464821
RH-Acked-by: Justin M. Forbes <jforbes@redhat.com>

Be sure to free resources when domain is destroyed.

Upstream Status: Accepted (CS 17441, 17533, 18585)

BZ: 464821

Signed-off-by: Xiaohui Xin <xiaohui.xin@intel.com>
Signed-off-by: Edwin Zhai <edwin.zhai@intel.com>
Signed-off-by: Don Dugger <donald.d.dugger@intel.com>

diff --git a/arch/x86/hvm/hvm.c b/arch/x86/hvm/hvm.c
index 89e02e9..6b414a6 100644
--- a/arch/x86/hvm/hvm.c
+++ b/arch/x86/hvm/hvm.c
@@ -275,6 +275,8 @@ void hvm_domain_relinquish_resources(struct domain *d)
 
 void hvm_domain_destroy(struct domain *d)
 {
+    if ( hvm_funcs.domain_destroy )
+        hvm_funcs.domain_destroy(d);
 }
 
 static int hvm_save_cpu_ctxt(struct domain *d, hvm_domain_context_t *h)
